docs_src/microservices/application/services/AppLLRPInventory/details/Behaviors.md

```diff
@@ -0,0 +1,237 @@
+- When it gets a stop request, it stops and/or disables ROSpecs, as necessary.
+
+
+## Important Limitations
+
+- Since this code makes use of various commands on the device service,
+  your devices must be registered with the LLRP device service
+  using device profiles with names matching
+  the `deviceCommands` and `deviceResources` it needs.
+  The full list [can be found below](#device-profile-requirements).
+- In particular, Impinj devices must be registered with the device service
+  using a profile that has an `enableImpinjExt` `deviceCommand`,
+  to `set` to a `deviceResource` representing a `CustomMessage`
+  that enables Impinj's custom extensions.
+  An example profile that meets these conditions
+  is available in the LLRP Device Service.
+- You can modify a Behavior at any time,
+  but doing so resets managed Readers' configurations,
+  and thus if they were reading, they will stop.
+- The current code uses only a single `default` Behavior that applies to all Readers,
+  and currently isn't persisted between restarts.
+  The code is written in a way to support multiple Behaviors,
+  but since there are many ways one can reasonably relate behaviors and devices,
+  extending this is left for future development
+  or as an exercise for the user.
+
+
+## Working with Behaviors
+
+To start all Readers reading with the current behavior,
+`POST` to the `/command/reading/start` endpoint:
+
+    curl -o- -X POST localhost:59711/api/v3/command/reading/start
+
+To stop reading,
+`POST` to the `/command/reading/stop` endpoint:
+
+    curl -o- -X POST localhost:59711/api/v3/command/reading/stop
+
+To view the `default` Behavior:
+
+    curl -o- localhost:59711/api/v3/behaviors/default
+
+```json
+{
+    "impinjOptions": {
+        "suppressMonza": false
+    },
+    "scanType": "Normal",
+    "duration": 0,
+    "power": {
+        "max": 3000
+    }
+}
+```
+
+To modify the `default` Behavior, `PUT` a new one at that endpoint.
+The new behavior completely replaces the old one.
+The following example uses `jq` to generate the `JSON` Behavior structure,
+the details of which are explained below.
+This particular Behavior enables a `Fast` scan at `30 dBm`:
+
+    curl -o- localhost:59711/api/v3/behaviors/default -XPUT \
+        --data @<(jq -n '{ScanType: "Fast", Power: {Max: 3000}}')
+
+
+If you attempt to set the Behavior to something that can't be supported
+all the Readers to which it should apply,
+you'll receive an error response, and the Behavior won't change:
+
+    curl -o- -XPUT localhost:59711/api/v3/behaviors/default \
+        --data @<(jq -n '{ScanType: "Fast"}')
+    
+    new behavior is invalid for "Speedway": target power (0.00 dBm)
+    is lower than the lowest supported (10.00 dBm): behavior cannot be satisfied
+
+## Supported Behavior Options
+- `ScanType` is a string which should be set to one of the following:
+    - `Fast` singulates tags as often as possible, with little regard for duplicates.
```

Comment:
```suggestion
    - `Fast` singulates tags as often as possible, with little regard for duplicates.
```
`singulates` typo ? simulates?

Reply:
Not simulates. @ajcasagrande can you help on the meaning here?

Reply:
singulates is correct, as it means:

> Singulation is a method by which an RFID reader identifies a tag with a specific serial number from a number of tags in its field. This is necessary because if multiple tags respond simultaneously to a query, they will jam each other.

Reply:
ok, i just could not find that word on the dictionary ; thanks for explanation.